File: Application/src/main/java/com/shareyourproxy/api/rx/command/eventcallback/GroupChannelsUpdatedEventCallback.java
package com.shareyourproxy.api.rx.command.eventcallback;

import android.os.Parcel;

import com.shareyourproxy.api.domain.model.Group;
import com.shareyourproxy.api.domain.model.User;
import com.shareyourproxy.app.EditGroupChannelsActivity.GroupEditType;

import java.util.HashSet;

/**
 * Group Name may have changed, channels updated.
 */
public class GroupChannelsUpdatedEventCallback extends UserEventCallback {
    private final static java.lang.ClassLoader CL =
        GroupChannelsUpdatedEventCallback.class.getClassLoader();
    public static final Creator<GroupChannelsUpdatedEventCallback> CREATOR =
        new Creator<GroupChannelsUpdatedEventCallback>() {
            @Override
            public GroupChannelsUpdatedEventCallback createFromParcel(Parcel in) {
                return new GroupChannelsUpdatedEventCallback(
                    (User) in.readValue(CL), (Group) in.readValue(CL),
                    (HashSet<String>) in.readValue(CL),(GroupEditType) in.readValue(CL));
            }

            @Override
            public GroupChannelsUpdatedEventCallback[] newArray(int size) {
                return new GroupChannelsUpdatedEventCallback[size];
            }
        };
    public final Group group;
    public final HashSet<String> channels;
    public final GroupEditType groupEditType;

    public GroupChannelsUpdatedEventCallback(
        User user, Group group, HashSet<String> channels, GroupEditType groupEditType) {
        super(user);
        this.group = group;
        this.channels = channels;
        this.groupEditType = groupEditType;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(group);
        dest.writeValue(channels);
        dest.writeValue(groupEditType);
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/command/eventcallback/GroupChannelsUpdatedEventCallback.kt
package com.shareyourproxy.api.rx.command.eventcallback

import android.os.Parcel
import android.os.Parcelable
import com.shareyourproxy.api.domain.model.Group
import com.shareyourproxy.api.domain.model.User
import com.shareyourproxy.util.Enumerations
import java.util.*

/**
 * Group Name may have changed, channels updated.
 */
internal class GroupChannelsUpdatedEventCallback(user: User, val oldGroup: Group, val group: Group, val channels: HashSet<String>, val groupEditType: Enumerations.GroupEditType) : UserEventCallback(user) {

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(oldGroup)
        dest.writeValue(group)
        dest.writeValue(channels)
        dest.writeValue(groupEditType)
    }

    companion object {
        private val CL = GroupChannelsUpdatedEventCallback::class.java.classLoader
        @JvmField val CREATOR: Parcelable.Creator<GroupChannelsUpdatedEventCallback> = object : Parcelable.Creator<GroupChannelsUpdatedEventCallback> {
            @Suppress("UNCHECKED_CAST")
            override fun createFromParcel(parcel: Parcel): GroupChannelsUpdatedEventCallback {
                return GroupChannelsUpdatedEventCallback(parcel.readValue(CL) as User, parcel.readValue(CL) as Group, parcel.readValue(CL) as Group, parcel.readValue(CL) as HashSet<String>, parcel.readValue(CL) as Enumerations.GroupEditType)
            }

            override fun newArray(size: Int): Array<GroupChannelsUpdatedEventCallback?> {
                return arrayOfNulls(size)
            }
        }
    }
}
